mixer/adapter/bypass/bypass.go

```diff
@@ -0,0 +1,302 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+//go:generate $GOPATH/src/istio.io/istio/bin/mixer_codegen.sh -f mixer/adapter/bypass/config/config.proto
+
+package bypass
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"go.uber.org/multierr"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+
+	"istio.io/api/mixer/adapter/model/v1beta1"
+	"istio.io/istio/mixer/adapter/bypass/config"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/template/checknothing"
+	"istio.io/istio/mixer/template/metric"
+	"istio.io/istio/mixer/template/quota"
+	"istio.io/istio/mixer/template/reportnothing"
+)
+
+// GetInfo returns the Info associated with this adapter implementation.
+func GetInfo() adapter.Info {
+	return adapter.Info{
+		Name:        "bypass",
+		Impl:        "istio.io/istio/mixer/adapter/bypass",
+		Description: "Calls gRPC backends via the inline adapter model (useful for testing)",
+		SupportedTemplates: []string{
+			checknothing.TemplateName,
+			reportnothing.TemplateName,
+			metric.TemplateName,
+			quota.TemplateName,
+		},
+		DefaultConfig: &config.Params{},
+		NewBuilder:    func() adapter.HandlerBuilder { return &builder{} },
+	}
+}
+
+type builder struct {
+	conn        *grpc.ClientConn
+	infraClient v1beta1.InfrastructureBackendClient
+
+	checknothingTypes  map[string]*checknothing.Type
+	reportnothingTypes map[string]*reportnothing.Type
+	metricTypes        map[string]*metric.Type
+	quotaTypes         map[string]*quota.Type
+
+	params *config.Params
+}
+
+var _ checknothing.HandlerBuilder = &builder{}
+var _ reportnothing.HandlerBuilder = &builder{}
+var _ metric.HandlerBuilder = &builder{}
+var _ quota.HandlerBuilder = &builder{}
+
+func (b *builder) SetCheckNothingTypes(checknothingTypes map[string]*checknothing.Type) {
+	b.checknothingTypes = checknothingTypes
+}
+
+func (b *builder) SetReportNothingTypes(reportnothingTypes map[string]*reportnothing.Type) {
+	b.reportnothingTypes = reportnothingTypes
+}
+
+func (b *builder) SetMetricTypes(metricTypes map[string]*metric.Type) {
+	b.metricTypes = metricTypes
+}
+
+func (b *builder) SetQuotaTypes(quotaTypes map[string]*quota.Type) {
+	b.quotaTypes = quotaTypes
+}
+
+func (b *builder) SetAdapterConfig(cfg adapter.Config) {
+	b.params = cfg.(*config.Params)
+}
+
+func (b *builder) Validate() (ce *adapter.ConfigErrors) {
+	if b.conn == nil {
```

Comment:
consider refactoring this block out so it can be used from other functions `build`